### PR TITLE
Change `NCASubject` to a function that returns a DataFrame

### PR DESCRIPTION
**Pull request by YingboMa**
Fix https://github.com/PumasAI/Pumas.jl/issues/878

### CODE REVIEW

**Comment by andreasnoack** [comment]
Please use a descriptive title and explain how the fix works. It makes it much easier to review the PR.

It looks like this makes the `NCAReport` constructor return a `DataFrame` which I don't think is a good idea. Is it still necessary to have an `NCAReport` struct or could `NCAreport`/`NCAReport` just be a function instead of a constructor.

**Comment by pkofod** [comment]
Maybe keep the type (if it's useful and plays a role), don't pipe it into `DataFrame`, but instead have a shallow `ncareport(...) = DataFrame(NCAReport)` function?

**Comment by pkofod** [comment]
LGTM

**Comment by andreasnoack** [comment]
Same here. Any thoughts on the Markdown code?

**Comment by vjd** [comment]
I think the DataFrame will render pretty well to markdown, so we can move that part out

File: src/nca/type.jl [hunk 1]
```diff
@@ -426,24 +421,12 @@ function NCAReport(pop::NCAPopulation; pred=nothing, normalize=nothing, auctype=
...
-# units go under the header
-to_dataframe(report::NCAReport) = convert(DataFrame, report)
-DataFrames.DataFrame(report::NCAReport) = to_dataframe(report)
-function Base.convert(::Type{DataFrame}, report::NCAReport)
-  #report.settings[:subject] && return DataFrame(map(x->[x], report.values))
-  hcat(report.values...)
-end
+to_dataframe(report) = (@warn("`to_dataframe` is removed. `NCAReport` now returns a DataFrame directly."); report)
 
+#=
```
Comment (andreasnoack) [inline src/nca/type.jl]: Then let's just delete the Markdown code and this should be good to go.

**Pull request merged**